runtimes/core/src/api/gateway/mod.rs

```diff
@@ -249,11 +259,15 @@ impl ProxyHttp for Gateway {
 
         upstream_request.set_uri(new_uri);
 
+        // Do we need to set the host header here?
+        // It means the upstream service won't be able to tell
+        // what the original Host header was, which is sometimes useful.
```

Comment:
it depends a bit on how the upstream works, if the upstream server handles multiple hosts or it should terminate tls in will need the host to figure out which certificate it should use

Reply:
another thing to consider regarding dns-based upstreams is that we resolve the dns, but just take the first ip, which might be fine depending on how the upstreams are configured